wgsl/scanner.cc

```diff
@@ -734,19 +769,30 @@ struct Scanner {
         } else {
           LOG("   non-template '>'");
           state.gt_is_tmpl.push_back(false);
+          // Pre-emptvely allow >= as a comparison operator:
+          // Skip over '=', if present.
+          lexer.match('=');
         }
         continue;
       }
 
+      // Pre-emptively allow the != operator.
+      // As a side effect, allow unary negation operator !
+      if (lexer.match('!')) {
+        lexer.match('=');
+        continue;
+      }
+
+      CodePoint was = lexer.peek();
       if (lexer.match_anyof({'(', '['})) {
-        LOG("   expr_depth++");
+        LOG("   %c expr_depth++", was);
```

Comment:
You might need to `static_cast<int>(was)`, depending on how picky the compiler wants to be.